Funky-Chickens/workFORCEmanagement
# GET all, GET one, POST functions for trainings

### I made the following changes:

- add ability to distill values returned from get trainings into data values array based on Jason's suggestion
- commented out portions temporarily to test get all and get one, post one training program, 
- modified get all trainings
- added full functionality for get one training and tested in psql
- add function for rendering form for training programs
- rearrange order of training route.js to avoid -conflict with routes that use req.params.id
- add POST function for training programs and test with postman
- merge most recent master and resolved merge conflicts
- retest all functions with postman and database, all working paths
- add post function for employees

### Reasons:
- to provide working functions to post, get one, get all trainings, post employee

### To Test Routes and Output:

- make sure you have your config/config.json set up before you start testing if necessary

#### POST a training :
1.  Open Postman app in chrome and your terminal.
2. Run ```npm start``` in the terminal
3.  Set the url to localhost:4000/training and the action to POST
4.  Make sure the body is set to JSON format, and put in an object like this:
{
	"name":"Your Fave Training Here",
	"start_date":"2017-11-25",
	"end_date":"2017-11-27",
	"max_attendees": 99,
	"createdAt":null,
	"updatedAt":null
}
5.  Hit send, and you should get a confirmation with an id assigned that looks like this:
{
    "id": 10,
    "name": "Your Fave Training Here",
    "start_date": "2017-11-25",
    "end_date": "2017-11-27",
    "max_attendees": 99,
    "createdAt": null,
    "updatedAt": null
}

#### POST employee:  Follow same basic procedure as above, just use the route ```localhost:4000/employees``` and use this format in postman:
{
    "first_name":"Matt",
    "last_name":"Smith",
    "dept_id": 4,
    "hire_date":"2006-04-25T23:59:03.244Z",
    "createdAt":null,
    "updatedAt":null
  }

#### GET ALL trainings:
1.  Open Postman app in chrome and your terminal.
2. Run ```npm start``` in the terminal
3.  Set the url to localhost:4000/training and the action to GET
4.  Hit send. 
5.  In postman, response should be the pug template that will eventually list all training programs.
6.  In your terminal, you should see this message:
SELECT "id", "name", "start_date", "end_date", "max_attendees" FROM "Trainings" AS "Training";
7.  Copy that statement and open a new git bash terminal.  Go to the database with ```psql``` command.
8.  Connect to the database with ```\c bangazon_workforce```.
9.  Test the output by pasting in the SQL select statement and hitting enter.  You should see all the available training programs listed in the table.

#### GET ONE training:  
1.  Follow the same procedure as for the GET all, but use the route ```localhost:4000/training/[id you put in]```, and you should see this in your terminal:
SELECT "id", "name", "start_date", "end_date", "max_attendees" FROM "Trainings" AS "Training"
WHERE "Training"."id" = '[id you put in]';
2.  Copy that statement and open a new git bash terminal.  Go to the database with ```psql``` command.
3.  Connect to the database with ```\c bangazon_workforce```.
4.  Test the output by pasting in the SQL select statement and hitting enter.  You should see the one  training program you selected.  

Refer to issue #9 , issue #10, issue #2
